runners/google-cloud-dataflow-java/examples-streaming/build.gradle

```diff
@@ -47,6 +47,9 @@ task windmillPreCommit(type: Test) {
      "--tempRoot=${gcsTempRoot}",
      "--runner=TestDataflowRunner",
      "--dataflowWorkerJar=${dataflowWorkerJar}",
+     // Provide job with a customizable worker jar.
+     // With legacy worker jar, containerImage is set to empty (i.e. to use the internal build).
+     // More context and discussions can be found in PR#6694.
```

Comment:
@Abacn I think this is the root cause of the issue.
By setting `workerHarnessContainerImage` to an empty string to pick up the non-versioned image as explained in https://github.com/apache/beam/pull/6694. It looks like this image ignores the java version.

@lukecwik can you help on how to select the proper container here ?

Reply:
Just wanted to confirm that removing the `workerHarnessContainerImage` and running with default image works.
![image](https://user-images.githubusercontent.com/2845540/227146807-4456a1dd-ea2a-45f5-9074-3e58845008a2.png)